crypto/provider_core.c

```diff
@@ -203,8 +203,8 @@ int ossl_provider_disable_fallback_loading(OSSL_LIB_CTX *libctx)
 {
     struct provider_store_st *store;
 
-    if ((store = get_provider_store(libctx)) != NULL) {
-        CRYPTO_THREAD_write_lock(store->lock);
+    if ((store = get_provider_store(libctx)) != NULL
+            || !CRYPTO_THREAD_write_lock(store->lock)) {
```

Comment:
This looks wrong.  It will modify the store without having the lock if the first condition is true.

Reply:
"Rich, have you met my friend DeMorgan?"  Sigh.  Fixed.